rllib/core/rl_trainer/tf/tf_rl_trainer.py

```diff
@@ -102,10 +101,32 @@ def __init__(
         tf1.enable_eager_execution()
 
         self._enable_tf_function = framework_hyperparameters.eager_tracing
+
+    @override(RLTrainer)
+    def build(self) -> None:
+        # TODO (Kourosh): Does strategy has to be an attribute here? if so it's very
+        # hidden to the user of this class that there is such an attribute.
+        if not self._distributed:
+            if self._use_gpu:
+                # mirrored strategy is typically used for multi-gpu training
+                # on a single machine, however we can use it for single-gpu
+                local_gpu = [f"GPU:{self._local_gpu_id}"]
+                self.strategy = tf.distribute.MirroredStrategy(devices=local_gpu)
```

Comment:
Can we define the variable `self.strategy` in the constructor and set it to some default there and overwrite here?

Reply:
I think self.strategy should be private.